From a17112eb59db344ad41ca262666872939de3d644 Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 17 Dec 2015 15:24:03 +0100
Subject: [PATCH 6/7] S2: Add settings for S2

This commit add/change/delete several things for the support of the S2:

* remove PMIC Support
* Change the names of the leds
* Change iomux settings for GPIOs
* Add NAND Support

Ticket #872
---
 arch/arm/boot/dts/imx6sx-s2.dts | 276 ++++++++++++----------------------------
 1 file changed, 84 insertions(+), 192 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-s2.dts b/arch/arm/boot/dts/imx6sx-s2.dts
index 5216720..e99734b 100644
--- a/arch/arm/boot/dts/imx6sx-s2.dts
+++ b/arch/arm/boot/dts/imx6sx-s2.dts
@@ -31,25 +31,29 @@
             gpios = <&gpio3 19 GPIO_ACTIVE_HIGH>;
             linux,default-trigger = "heartbeat";
         };
+        led {
+            label = "Red_led";
+            gpios = <&gpio3 20 GPIO_ACTIVE_HIGH>;
+        };
     };
 
-	regulators {
-		compatible = "simple-bus";
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		reg_5p0: regulator@4 {
-			compatible = "regulator-fixed";
-			reg = <4>;
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_5pe>;
-			regulator-name = "5p0";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-			gpio = <&gpio2 12 GPIO_ACTIVE_HIGH>;
-			regulator-boot-on;
-		};
-	};
+	/*regulators {*/
+		/*compatible = "simple-bus";*/
+		/*#address-cells = <1>;*/
+		/*#size-cells = <0>;*/
+
+		/*reg_5p0: regulator@4 {*/
+			/*compatible = "regulator-fixed";*/
+			/*reg = <0>;*/
+			/*pinctrl-names = "default";*/
+			/*pinctrl-0 = <&pinctrl_5pe>;*/
+			/*regulator-name = "5p0";*/
+			/*regulator-min-microvolt = <5000000>;*/
+			/*regulator-max-microvolt = <5000000>;*/
+			/*gpio = <&gpio2 12 GPIO_ACTIVE_LOW>;*/
+			/*regulator-always-on;*/
+		/*};*/
+	/*};*/
 };
 
 &cpu0 {
@@ -74,6 +78,7 @@
 &gpc {
 	/* use ldo-bypass, u-boot will check it and configure */
 	fsl,ldo-bypass = <0>;
+	fsl,wdog-reset = <1>; /* watchdog select of reset source */
 };
 
 &uart1 {
@@ -95,11 +100,16 @@
 	status = "okay";
 };
 
+&gpmi {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_gpmi_nand_1>;
+    status = "okay";
+    nand-on-flash-bbt;
+};
+
 &flexcan2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexcan2>;
-	trx-stby-gpio = <&gpio1 02 GPIO_ACTIVE_LOW>;
-	trx_wakeup_gpio = <&gpio1 03 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 };
 
@@ -109,7 +119,7 @@
 };
 
 &usbotg2 {
-	dr_mode = "host";
+	dr_mode = "peripheral";
 	status = "okay";
 };
 
@@ -142,143 +152,51 @@
 		compatible = "ti,lp5562";
         label = "LED1";
 		reg = <0x30>;
-		clock-mode = /bits/8 <1>;
+		clock-mode = /bits/8 <2>;
 
 		chan0 {
-			chan-name = "R";
+			chan-name = "R1";
 			led-cur = /bits/ 8 <0x20>;
 			max-cur = /bits/ 8 <0x60>;
 		};
 		chan1 {
-			chan-name = "G";
+			chan-name = "G1";
 			led-cur = /bits/ 8 <0x20>;
 			max-cur = /bits/ 8 <0x60>;
 		};
 
 		chan2 {
-			chan-name = "B";
+			chan-name = "B1";
 			led-cur = /bits/ 8 <0x20>;
 			max-cur = /bits/ 8 <0x60>;
 		};
 	};
-	led2: lp5562@31 {
+
+	led2: lp5562@32 {
 		compatible = "ti,lp5562";
         label = "LED2";
-		reg = <0x31>;
-		clock-mode = /bits/8 <1>;
+		reg = <0x32>;
+		clock-mode = /bits/8 <2>;
 
 		chan0 {
-			chan-name = "R";
+			chan-name = "R2";
 			led-cur = /bits/ 8 <0x20>;
 			max-cur = /bits/ 8 <0x60>;
 		};
 		chan1 {
-			chan-name = "G";
+			chan-name = "G2";
 			led-cur = /bits/ 8 <0x20>;
 			max-cur = /bits/ 8 <0x60>;
 		};
 
 		chan2 {
-			chan-name = "B";
+			chan-name = "B2";
 			led-cur = /bits/ 8 <0x20>;
 			max-cur = /bits/ 8 <0x60>;
 		};
 	};
 };
 
-&i2c4 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c4>;
-	status = "okay";
-
-	pmic: pfuze200@08 {
-		compatible = "fsl,pfuze200";
-		reg = <0x08>;
-
-		regulators {
-			sw1a_reg: sw1ab {
-				regulator-min-microvolt = <300000>;
-				regulator-max-microvolt = <1875000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3a {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1975000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3b_reg: sw3b {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1975000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
-			};
-
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vgen1_reg: vgen1 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
-
-			vgen2_reg: vgen2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
-
-			vgen3_reg: vgen3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-			};
-
-			vgen4_reg: vgen4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen5_reg: vgen5 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen6_reg: vgen6 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-		};
-	};
-};
-
 &ecspi3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_ecspi3 &pinctrl_ecspi3_cs>;
@@ -288,77 +206,54 @@
 	status = "okay";
 };
 
-/*&gpmi {*/
-	/*pinctrl-names = "default";*/
-	/*pinctrl-0 = <&pinctrl_gpmi_nand_1>;*/
-	/*status = "okay";*/
-	/*nand-on-flash-bbt;*/
-/*};*/
-
 &iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_can_gpios &pinctrl_pctrl &pinctrl_antenna_sel>;
+	pinctrl-0 = <&pinctrl_hog>;
 
 	imx6x-score {
-		pinctrl_5pe: power_5P0_enable {
-			fsl,pins = <
-				MX6SX_PAD_KEY_COL2__GPIO2_IO_12		0x4018
-			>;
-		};
-
-		pinctrl_pctrl: power-ctrl {
-			fsl,pins = <
-				/* WIFI_PWR_E */
-				MX6SX_PAD_LCD1_HSYNC__GPIO3_IO_26	0x4018
-				/* GNSS_PWR_E */
-				MX6SX_PAD_LCD1_DATA08__GPIO3_IO_9	0x4018
-				/* GNSS_RESET */
-				MX6SX_PAD_LCD1_DATA00__GPIO3_IO_1	0x4018
-				/* kill_4P */
-				MX6SX_PAD_LCD1_DATA10__GPIO3_IO_11	0x4018
-				/* LED_E */
-				MX6SX_PAD_LCD1_DATA09__GPIO3_IO_10	0x4018
-				/* GSM_RES */
-				MX6SX_PAD_GPIO1_IO09__GPIO1_IO_9	0x4018
-				/* GSM_IGN */
-				MX6SX_PAD_LCD1_DATA14__GPIO3_IO_15	0x4018
-			>;
-		};
-
-		pinctrl_antenna_sel: antenna-sel {
-			fsl,pins = <
-				/* WIFI_ANTENNA_SEL */
-				MX6SX_PAD_LCD1_DATA15__GPIO3_IO_16	0x4018
-				/* GNSS_ANTENNA_SEL */
-				MX6SX_PAD_LCD1_RESET__GPIO3_IO_27	0x4018
-				/* GSM_ANTENNA_SEL */
-				MX6SX_PAD_LCD1_ENABLE__GPIO3_IO_25	0x4018
-			>;
-		};
-
-		/*pinctrl_gpmi_nand_1: gpmi-nand-1 {*/
+		/*pinctrl_5pe: power_5P0_enable {*/
 			/*fsl,pins = <*/
-				/*MX6SX_PAD_NAND_CLE__RAWNAND_CLE         0xb0b1*/
-				/*MX6SX_PAD_NAND_ALE__RAWNAND_ALE         0xb0b1*/
-				/*MX6SX_PAD_NAND_WP_B__RAWNAND_WP_B       0xb0b1*/
-				/*MX6SX_PAD_NAND_READY_B__RAWNAND_READY_B 0xb000*/
-				/*MX6SX_PAD_NAND_CE0_B__RAWNAND_CE0_B     0xb0b1*/
-				/*MX6SX_PAD_NAND_RE_B__RAWNAND_RE_B       0xb0b1*/
-				/*MX6SX_PAD_NAND_WE_B__RAWNAND_WE_B       0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA00__RAWNAND_DATA00   0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA01__RAWNAND_DATA01   0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA02__RAWNAND_DATA02   0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA03__RAWNAND_DATA03   0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA04__RAWNAND_DATA04   0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA05__RAWNAND_DATA05   0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA06__RAWNAND_DATA06   0xb0b1*/
-				/*MX6SX_PAD_NAND_DATA07__RAWNAND_DATA07   0xb0b1*/
+				/*MX6SX_PAD_KEY_COL2__GPIO2_IO_12		0x80*/
 			/*>;*/
 		/*};*/
 
-		pinctrl_gsm_state: gsm-state {
+        pinctrl_hog: hoggrp {
+			fsl,pins = <
+				MX6SX_PAD_LCD1_HSYNC__GPIO3_IO_26	0x17059
+				MX6SX_PAD_LCD1_DATA08__GPIO3_IO_9	0xc9
+				MX6SX_PAD_LCD1_DATA00__GPIO3_IO_1	0xc9
+				MX6SX_PAD_LCD1_DATA10__GPIO3_IO_11	0x17059
+				MX6SX_PAD_LCD1_DATA09__GPIO3_IO_10	0x17059
+				MX6SX_PAD_GPIO1_IO09__GPIO1_IO_9	0xc9
+				MX6SX_PAD_LCD1_DATA14__GPIO3_IO_15	0xc9
+				MX6SX_PAD_LCD1_DATA15__GPIO3_IO_16	0x17059
+				MX6SX_PAD_LCD1_RESET__GPIO3_IO_27	0x8f
+				MX6SX_PAD_LCD1_ENABLE__GPIO3_IO_25	0x8f
+				MX6SX_PAD_SD3_DATA6__GPIO7_IO_8		0x17059
+				MX6SX_PAD_GPIO1_IO08__GPIO1_IO_8	0x17059
+				MX6SX_PAD_GPIO1_IO03__GPIO1_IO_3	0x17059
+				MX6SX_PAD_KEY_COL2__GPIO2_IO_12		0xc9
+                MX6SX_PAD_GPIO1_IO02__GPIO1_IO_2	0xc9
+			>;
+        };
+
+		pinctrl_gpmi_nand_1: gpmi-nand-1 {
 			fsl,pins = <
-				MX6SX_PAD_SD3_DATA6__GPIO7_IO_8		0x40
+				MX6SX_PAD_NAND_CLE__RAWNAND_CLE         0xb0b1
+				MX6SX_PAD_NAND_ALE__RAWNAND_ALE         0xb0b1
+				MX6SX_PAD_NAND_WP_B__RAWNAND_WP_B       0xb0b1
+				MX6SX_PAD_NAND_READY_B__RAWNAND_READY_B 0xb000
+				MX6SX_PAD_NAND_CE0_B__RAWNAND_CE0_B     0xb0b1
+				MX6SX_PAD_NAND_RE_B__RAWNAND_RE_B       0xb0b1
+				MX6SX_PAD_NAND_WE_B__RAWNAND_WE_B       0xb0b1
+				MX6SX_PAD_NAND_DATA00__RAWNAND_DATA00   0xb0b1
+				MX6SX_PAD_NAND_DATA01__RAWNAND_DATA01   0xb0b1
+				MX6SX_PAD_NAND_DATA02__RAWNAND_DATA02   0xb0b1
+				MX6SX_PAD_NAND_DATA03__RAWNAND_DATA03   0xb0b1
+				MX6SX_PAD_NAND_DATA04__RAWNAND_DATA04   0xb0b1
+				MX6SX_PAD_NAND_DATA05__RAWNAND_DATA05   0xb0b1
+				MX6SX_PAD_NAND_DATA06__RAWNAND_DATA06   0xb0b1
+				MX6SX_PAD_NAND_DATA07__RAWNAND_DATA07   0xb0b1
 			>;
 		};
 
@@ -371,12 +266,9 @@
 
 		pinctrl_can_gpios: can-gpios {
 			fsl,pins = <
-				/* CAN_TERMINATE */
-				MX6SX_PAD_GPIO1_IO08__GPIO1_IO_8	0x17059
 				/* CAN_STANDBY */
-				MX6SX_PAD_GPIO1_IO02__GPIO1_IO_2	0x17059
-				/* CAN_WAKEUP */
-				MX6SX_PAD_GPIO1_IO03__GPIO1_IO_3	0x17059
+				/*MX6SX_PAD_GPIO1_IO02__GPIO1_IO_2	0x17059*/
+				MX6SX_PAD_GPIO1_IO02__GPIO1_IO_2	0xc9
 			>;
 		};
 
@@ -418,8 +310,8 @@
 
 		pinctrl_uart3: uart3grp {
 			fsl,pins = <
-				MX6SX_PAD_SD3_DATA5__UART3_RX		0x1b0b1
-				MX6SX_PAD_SD3_DATA4__UART3_TX		0x1b0b1
+				MX6SX_PAD_SD3_DATA5__UART3_TX		0x1b0b1
+				MX6SX_PAD_SD3_DATA4__UART3_RX		0x1b0b1
 			>;
 		};
 
-- 
1.9.1

